src/azul/bigquery_reservation.py

```diff
@@ -78,53 +91,80 @@ def _reservation_parent_path(self) -> str:
         return self._client.common_location_path(project=self._project,
                                                  location='US')
 
-    def has_active_slots(self) -> bool:
+    def state(self) -> SlotState:
```

Comment:
Nothing against use of type-safe enums but this can be modeled more concisely as is a tri-state boolean (yes,no, don't know) without losing expressiveness,

```suggestion
    @property
    def is_active(self) -> Optional[bool]:
```